pkg/webhookconfig/registration.go

```diff
@@ -746,18 +742,39 @@ func (wrc *Register) updateResourceValidatingWebhookConfiguration(nsSelector map
 		if !ok {
 			return errors.Wrapf(err, "type mismatched, expected map[string]interface{}, got %T", webhooksUntyped[i])
 		}
+		{
+			currentSelector, _, err := unstructured.NestedMap(webhook, "objectSelector")
+			if err != nil {
+				return errors.Wrapf(err, "unable to get validatingWebhookConfigurations.webhooks["+fmt.Sprint(i)+"].objectSelector")
+			}
 
-		currentSelector, _, err := unstructured.NestedMap(webhook, "namespaceSelector")
-		if err != nil {
-			return errors.Wrapf(err, "unable to get validatingWebhookConfigurations.webhooks["+fmt.Sprint(i)+"].namespaceSelector")
-		}
-
-		if !reflect.DeepEqual(nsSelector, currentSelector) {
-			webhookChanged = true
+			if expectedSelector, err := convertLabelSelector(webhookCfg.ObjectSelector); err != nil {
+				return err
+			} else {
+				if !reflect.DeepEqual(expectedSelector, currentSelector) {
+					webhookChanged = true
+				}
+				if err = unstructured.SetNestedMap(webhook, expectedSelector, "objectSelector"); err != nil {
+					return errors.Wrapf(err, "unable to set validatingWebhookConfigurations.webhooks["+fmt.Sprint(i)+"].objectSelector")
+				}
+			}
```

Comment:
Can we extract this logic to a separate method and re-use it for both selectors?

Reply:
Sure, I'll rework this.

Reply:
Done, created the `configureSelector` helper.